src/blob-uploader.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+'use strict'
+
+const privateVars = new WeakMap()
+
+function blobUploader (apiUrl /* :string */) {
+  if (!apiUrl) {
+    throw new TypeError('blobUploader expects a api URL during instantiation')
+  }
+  privateVars.set(this, {
+    uri: apiUrl
+  })
+}
+
+blobUploader.prototype.uploadBlob = function (
+  blob /*: Blob */
+) /* :Promise<number> */ {
+  if (!blob) {
+    return Promise.reject(new Error('blob argument not provided'))
+  }
+
+  if (!privateVars || !privateVars.get(this)) {
+    return Promise.reject(new Error('blobUploader uri not configured'))
+  }
+  const vars = privateVars.get(this)
+  if (!vars || !vars.hasOwnProperty('uri')) {
+    return Promise.reject(new Error('blobUploader uri not configured'))
+  }
+
+  const request = new Request(vars.uri, {
+    method: 'POST',
+    mode: 'cors'
+  })
+
+  return fetch(request)
+    .then((response) => {
+      if (!response.ok) {
+        return Promise.reject(new Error('Error calling blob api service: ' + response.status + ' ' + response.statusText))
+      }
+      return response.json()
+    })
+    .then((apiResponse) => {
+      this._uploadToS3(blob, apiResponse.putUrl)
+      return apiResponse.id
+    })
+    .catch((err) => Promise.reject(new Error('Error calling blob api service: ' + JSON.stringify(err))))
```

Comment:
`JSON.stringify(err)` will always return `{}`

See https://www.npmjs.com/package/serialize-error

I would instead just use `'' + err` as the `Error.prototype.toString()` returns `err.message`